test/component/profile_page_test.dart

```diff
@@ -66,23 +64,75 @@ void main() {
     await tester.pumpAndSettle();
 
     // Check that badges are displayed
-    final infoRow = find.byKey(InfoCard.cardKey);
-    expect(infoRow, findsWidgets);
+    final badgeCard = find.byKey(InfoCardBadge.cardKey);
+    expect(badgeCard, findsWidgets);
+
+    final postCard = find.byKey(InfoCardBadge.cardKey);
+    expect(postCard, findsWidgets);
+
+    final commentCard = find.byKey(InfoCardBadge.cardKey);
+    expect(commentCard, findsWidgets);
```

Comment:
I just think that you used the wrong keys here:
```suggestion
    final postCard = find.byKey(InfoCardPost.cardKey);
    expect(postCard, findsWidgets);

    final commentCard = find.byKey(InfoCardComment.cardKey);
    expect(commentCard, findsWidgets);
```